.github/workflows/image-kairos-ubuntu.yml

```diff
@@ -0,0 +1,57 @@
+---
+name: Build Kairos-Ubuntu
+
+on: # yamllint disable-line rule:truthy
+  push:
+    branches:
+      - main
+    paths:
+      - .github/workflows/image-kairos-ubuntu.yml
+      - images/kairos-ubuntu/**
+    tags:
+      - v*-kairos-ubuntu
+  pull_request:
+    branches:
+      - main
+    paths:
+      - .github/workflows/image-kairos-ubuntu.yml
+      - images/kairos-ubuntu/**
+  release:
+    types:
+      - created
+
+jobs:
+  build-container:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@74a5d142397b4f367a81961eba4e8cd7edddf772 # v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      # Note: this must start with a number for the build to work correctly!
+      - name: Compute Version Tag
+        env:
+          BRANCH: ${{ github.ref_type == 'tag' && github.ref_name || github.event_name != 'pull_request' && format('{0}.{1}', github.actor_id, github.job) || format('{0}-pr', github.event.number) }}
```

Comment:
I follow the 'use the tag name as the branch name if it's a tag' logic, but I don't follow the rest

Reply:
The Kairos factory requires the version to be numeric at the start, so I just picked something that should be always incrementing on the off chance we decide to publish anything other than tagged releases.  We want CI to run to make sure it can build, but we don't actually care about the version number so I picked something somewhat arbitrary for the other cases.

We might need to iterate on this, but it's a start!